homeassistant/components/knocki/config_flow.py

```diff
@@ -0,0 +1,66 @@
+"""Config flow for Knocki integration."""
+
+from __future__ import annotations
+
+from typing import Any
+
+from knocki import KnockiClient, KnockiConnectionError
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_PASSWORD, CONF_TOKEN, CONF_USERNAME
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN, LOGGER
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+class KnockiConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Knocki."""
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            session = async_get_clientsession(self.hass)
+            client = KnockiClient(session=session)
+            try:
+                token_response = await client.login(
+                    user_input[CONF_USERNAME], user_input[CONF_PASSWORD]
+                )
+            except KnockiConnectionError:
+                errors["base"] = "cannot_connect"
+            except Exception:  # noqa: BLE001
+                LOGGER.exception("Error logging into the Knocki API")
+                errors["base"] = "unknown"
+            else:
+                await self.async_set_unique_id(token_response.user_id)
+                self._abort_if_unique_id_configured()
+                client.token = token_response.token
+                try:
+                    await client.link()
```

Comment:
if it throws with the exact same reasons why not move this to the previous `try:` and save some duplicate code?

Reply:
Because this shouldn't be called if the unique_id is already configured, but moving the unique_id setting and abort to the try block felt strange as well